File: app/src/main/java/com/example/influencer/Data/Preferences/DataStore.java
package com.example.influencer.Data.Preferences;

import android.content.Context;

import androidx.datastore.preferences.core.Preferences;
import androidx.datastore.preferences.rxjava3.RxPreferenceDataStoreBuilder;
import androidx.datastore.rxjava3.RxDataStore;

public class DataStore {
    //https://www.notion.so/DataStore-5ff1d598536947498542c292ae3f2702
    private static final String PREFERENCES_NAME = "user_preferences";
    private static RxDataStore<Preferences> dataStoreInstance;

    public static  RxDataStore<Preferences> getDataStoreInstance(Context context){
        if(dataStoreInstance == null){
            dataStoreInstance = new RxPreferenceDataStoreBuilder(context,PREFERENCES_NAME).build();
        }

        return dataStoreInstance;
    }

}

File: app/src/main/java/com/example/influencer/Core/Data/Preferences/DataStore.kt
package com.example.influencer.Core.Data.Preferences;

import android.content.Context;

import androidx.datastore.preferences.core.Preferences;
import androidx.datastore.preferences.rxjava3.RxPreferenceDataStoreBuilder;
import androidx.datastore.rxjava3.RxDataStore;

import javax.inject.Inject;
import javax.inject.Singleton;

import dagger.hilt.android.qualifiers.ApplicationContext;

@Singleton
public class DataStore {
    //https://www.notion.so/DataStore-5ff1d598536947498542c292ae3f2702
    private static final String PREFERENCES_NAME = "user_preferences";
    private final RxDataStore<Preferences> dataStoreInstance;

    @Inject
    public DataStore(@ApplicationContext Context context) {
        dataStoreInstance = new RxPreferenceDataStoreBuilder(context, PREFERENCES_NAME).build();
    }

    public RxDataStore<Preferences> getDataStoreInstance() {
        return dataStoreInstance;
    }
}
